src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_INCREASE = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public ArrayList(int defaultCapacity) {
+        values = (T[]) new Object[defaultCapacity];
+        size = 0;
+    }
+
+    private T[] letGrow() {
```

Comment:
let’s move private methods to the bottom of the file, or at least after the place where we call them